src/ui/public/persisted_state/__tests__/persisted_state.js

```diff
@@ -0,0 +1,536 @@
+      });
+    });
+  });
+
+  describe('child state creation', function () {
+    var childState;
+
+    it('should not append the child state to the parent, without parent value', function () {
+      var childIndex = 'odd.keyname[]';
+      var persistedState = new PersistedState();
+      childState = persistedState.createChild(childIndex);
+
+      // parent state should contain the child and its original state
+      expect(persistedState.get()).to.not.have.property(childIndex);
+    });
+
+    it('should not append the child state to the parent, with parent value', function () {
+      var childIndex = 'i can haz child';
+      var childStateValue = {};
+      var persistedStateValue = { original: true };
+      var persistedState = new PersistedState(persistedStateValue);
+      childState = persistedState.createChild(childIndex);
+
+      // child state should be empty, we didn't give it any default data
+      expect(childState.get()).to.be(undefined);
+
+      // parent state should contain the child and its original state value
+      expect(persistedState.get()).to.not.have.property(childIndex);
+    });
+
+    it('should append the child state to the parent, with parent and child values', function () {
+      var childIndex = 'i can haz child';
+      var childStateValue = { tacos: 'yes please' };
+      var persistedStateValue = { original: true };
+      var persistedState = new PersistedState(persistedStateValue);
+      childState = persistedState.createChild(childIndex, childStateValue);
+
+      // child state should be empty, we didn't give it any default data
+      expect(childState.get()).to.eql(childStateValue);
+
+      // parent state should contain the child and its original state value
+      var parentState = persistedState.get();
+      expect(parentState).to.have.property('original', true);
+      expect(parentState).to.have.property(childIndex);
+      expect(parentState[childIndex]).to.eql(childStateValue);
+    });
+  });
+
+  describe('deep child state creation', function () {
+    it('should delegate get/set calls to parent state', function () {
+      var children = [{
+        path: 'first*child',
+        value: { first: true, second: false }
+      }, {
+        path: 'second child',
+        value: { first: false, second: true }
+      }];
+      var persistedStateValue = { original: true };
+      var persistedState = new PersistedState(persistedStateValue);
+
+      // first child is a child of the parent persistedState
+      children[0].instance = persistedState.createChild(children[0].path, children[0].value);
+      // second child is a child of the first child
+      children[1].instance = children[0].instance.createChild(children[1].path, children[1].value);
+
+      // second child getter should only return second child value
+      expect(children[1].instance.get()).to.eql(children[1].value);
+
+      // parent should contain original props and first child path, but not the second child path
```

Comment:
I think this would be a lot clearer if it just tested that `parentState` was `deep.eql()` to an object literal. Otherwise I don't think this test checks to see that the value of children[1] is in the `parentState`.